jumpy/_custom_fns.py

```diff
@@ -0,0 +1,106 @@
+"""Custom implemented functions."""
+
+from __future__ import annotations
+
+from typing import Any, Callable, Sequence, TypeVar
+
+import numpy as onp
+
+from jumpy import is_jax_installed, ndarray
+from jumpy.core import is_jitted, which_dtype, which_np
+
+try:
+    import jax
+    import jax.numpy as jnp
+except ImportError:
+    jax, jnp = None, None
+
+
+F = TypeVar("F", bound=Callable)
+
+
+def array(object: Any, dtype=None) -> ndarray:
+    """Creates an array given a list."""
+    if dtype is None:
+        try:
+            np = which_np(*object)
+        except TypeError:
+            np = which_np(object)  # object is not iterable (e.g. primitive type)
+    else:
+        np = which_dtype(dtype)
+
+    return np.array(object, dtype)
```

Comment:
Why is array placed in `_custom_fns` instead of `_factory_fns`?

Reply:
I just realised there is an issue with the plan of making this a drop-in replacement of jax with numpy support is that a lot of the time, we are calling functions in `jax.numpy` not `jax` but in jumpy they are `jumpy.XX`.

Therefore, if our motivation is for this to be a true drop-in replacement of jax we need a new file for numpy if that makes sense

Reply:
It's a bit double, but I think the benefit of having a clear defined API is worth it.

The style you recommend looks good to me!

Reply:
Ok, I will restructure this PR for that.